### PR TITLE
Support fetch/update chat from DB

### PR DESCRIPTION
**Pull request by junkaiman**
### Summary
Support fetch/update chat from DB

### Screenshot
image

image

### Checklist
- [x] My code follows ESLint rules and passes all tests.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by ZhuoyangM — COMMENTED** [review]
I've merged the changes from the resizable components branch to dev branch. Please pull from the updated dev branch and make corresponding changes and submit a pull request again!

File: insta-ui/src/components/ChatInterface/index.tsx
Comment (ZhuoyangM) [inline insta-ui/src/components/ChatInterface/index.tsx:1]: How about getting the chat instance only once and storing it in a local state? So you won't have to fetch it every time sending a message.

File: insta-ui/src/lib/db.ts
Comment (ZhuoyangM) [inline insta-ui/src/lib/db.ts:1]: Now the chats are sorted by lastModified time, so every time I send new messages to a chat it gets popped up to the top. In ChatGPT chats are sorted by createdTime which I think is a cleaner way to organize chats. Just an advice

File: insta-ui/src/components/ChatInterface/index.tsx
Comment (ZhuoyangM) [reply insta-ui/src/components/ChatInterface/index.tsx:1]: kk

File: insta-ui/src/app/page.tsx [hunk 1]
```diff
@@ -1,5 +1,3 @@
 import Main from "./c/[slug]/page";
 
-export default function Home() {
-  return <Main />;
-}
```
Comment (ZhuoyangM) [inline insta-ui/src/app/page.tsx:5]: dscsacas

File: insta-ui/src/app/page.tsx [hunk 2]
```diff
@@ -1,5 +1,3 @@
 import Main from "./c/[slug]/page";
```
Comment (junkaiman) [inline insta-ui/src/app/page.tsx:1]: hihihi

**Comment by junkaiman** [comment]
* TODO: use shallow routing to prevent unnecessary page refresh between chats. Major changes needed on folder structure.

**Review by ZhuoyangM — APPROVED** [review]

**Pull request merged**